### DISCUSSION TITLE
docs.compressx.app to compressx.app/docs

### DISCUSSION THREAD

**Discussion by dqhieu** [vercel/community#6571] [Help]
Hi everyone,

So I just published my GitBook document to https://docs.compressx.app
This works great, but I also want it to be accessible via https://compressx.app/docs
My friend was able to achieve this by using Cloudflare proxy worker. For me, I asked ChatGPT but it didn't give me any working solution with Vercel. 
Was anyone able to achieve this?

**Comment by versecafe** [comment]
@dqhieu middleware rewrite could be a simple way to just rewrite everything in `/docs` to `https://docs.comressx.app` like this

`middleware.ts`
```ts
import { NextRequest, NextResponse } from "next/server";

export const config = {
  matcher: [
    /*
     * Match all request paths except for the ones starting with:
     * - api (API routes)
     * - _next/static (static files)
     * - _next/image (image optimization files)
     * - favicon.ico (favicon file)
     */
    {
      source: "/((?!api|_next/static|_next/image|favicon.ico).*)",
      missing: [
        { type: "header", key: "next-router-prefetch" },
        { type: "header", key: "purpose", value: "prefetch" },
      ],
    },
  ],
};

export function middleware(request: NextRequest) {
  // gitbook docs redirect
  if (request.nextUrl.pathname.startsWith("/docs")) {
    const path: string = request.nextUrl.pathname.replace("/docs", "");
    return NextResponse.rewrite(new URL(path, "https://docs.compressx.app"));
  }
}
```

Documentation: https://nextjs.org/docs/app/api-reference/next-config-js/rewrites

example of it functioning in a NextJS app with the given middleware.ts

<img width="1384" alt="image" src="https://github.com/vercel/community/assets/147033096/b8d2913f-8175-41e0-9730-b69e5c966526">
<img width="1385" alt="image" src="https://github.com/vercel/community/assets/147033096/7a2ee81d-0b94-4cb9-825c-34921d172c08">
